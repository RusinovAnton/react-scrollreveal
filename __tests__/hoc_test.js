import React from 'react'
import withScrollReveal from '../src/index'
import { isElement } from 'react-dom/test-utils'
import ShallowRenderer from 'react-test-renderer/shallow'
const renderer = new ShallowRenderer();

const MyComponent = () => {
  return (
    <div>
      <p>Lorem ipsum dolor sit amet.</p>
      <p>Lorem ipsum dolor sit amet.</p>
      <p>Lorem ipsum dolor sit amet.</p>
    </div>
  )
};

describe('HOC', function () {
  it('should take object or array of objects of defined shape as parameter', function () {
    expect(withScrollReveal({ selector: '.item', options: { reset: true }, interval: 300 })).not.toThrow();
    expect(withScrollReveal([
      { selector: '.item', options: { reset: true }, interval: 300 },
      { selector: '.item', options: { reset: true } },
    ])).not.toThrow();
  });

  it('should return wrapped React component with additional props', function () {
    const WrappedComponent = withScrollReveal({
      selector: '.item',
      options: { reset: true },
      interval: 300
    })(MyComponent);
    const component = renderer.render(<WrappedComponent/>);
    const result = renderer.getRenderOutput();

    expect(isElement(component)).toBe(true);
    expect(component).toMatchSnapshot();
    expect(typeof result.props.animationContainerReference).toBe('function');
    expect(typeof result.props.destroyRevealAnimation).toBe('function');
    expect(typeof result.props.refreshRevealAnimation).toBe('function');
  });
});
