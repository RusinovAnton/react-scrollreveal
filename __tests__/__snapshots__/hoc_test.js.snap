// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HOC should return wrapped React component with additional props 1`] = `
<MyComponent
  animationContainerReference={[Function]}
  destroyRevealAnimation={[Function]}
  refreshRevealAnimation={[Function]}
/>
`;
